app/javascript/packs/applications.js

```diff
@@ -1,5 +1,10 @@
-import { initialize as planSelector } from '../src/Applications/plan_selector'
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { PlanSelector } from '../src/Common/PlanSelector'
 
 document.addEventListener('DOMContentLoaded', () => {
-  window.planSelector = planSelector
+  const node = document.getElementById('plan_selector')
+  const { plans } = JSON.parse(node.getAttribute('data'))
+
+  ReactDOM.render(<PlanSelector plans={plans} />, node)
```

Comment:
This is breaking CircleCI beacuse of Capybara:
>And I go to the provider side create application page for "bob"                                # features/step_definitions/web_steps.rb:22
>      {:line_number=>25553, :message=>"SyntaxError: Invalid character: '`'", :source=>"http://admin.foo.example.com:57371/packs-test/applications-ae1a728d25e2326ca7af.js"} (Capybara::Webkit::JavaScriptError)
That means we can't use `content_tag` after all?

Reply:
It is reading non-existing attribute as as mentioned in https://github.com/3scale/porta/pull/365#discussion_r238346682

Reply:
And since you are using `ReactDOM`, you'll need to import `core-js/fn/set` and `import 'core-js/es6/map'`. It uses these two extensible in its code

Reply:
@mikz the attribute is actually working just fine, i'll change it shouldn't be _plans_ as you point out, but just sayin'.

Reply:
@josemigallas I don't think `node.getAttribute('data')` is working just fine, because to me it looks like it fails to parse the JSON and throws an exception.

`content_tag :div, data: { plans: 'foo' }` creates `<div data-plans='foo'/>` as you can see in the generated HTML.

Reply:
This is what's being generated when running porta locally:
```html
<div
  id="plan_selector"
  data="{"plans":[{"id":532,"name":"Test"},{"id":533,"name":"Premiun"}]}">
</div>
```
But anyway, the failure in CircleCI is due to some syntax error in a pack file and this is happening regardless of the data attribute. I haven't figured out yet what is causing this, but it starts in 70eb04c3505593ca651d3f0764a6aac4d79dcb91.

Reply:
Interesting, there is something off.

The documentation says it works (for tag, but content_tag uses the same methods internally
).
https://api.rubyonrails.org/v4.2.11/classes/ActionView/Helpers/TagHelper.html#method-i-tag

Can you use `tag` instead of `content_tag`?

Also it works just fine in the `bin/rails console`:

```ruby
helper.tag :div, data: { foo: 'bar' }
"<div data-foo=\"bar\" />"

helper.content_tag :div, '', data: { foo: 'bar' }
"<div data-foo=\"bar\"></div>"
```

Reply:
Ah! I think the problem was the `to_json` in a bad place. It's `data: { plans: ... }.to_json()` whereas should be `data: { plans: (...).to_json() }`, isn't it?

Reply:
Yes.